lib/game.rb

```diff
@@ -0,0 +1,97 @@
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player1 = Player::Human.new("X"), player2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player1
+    @player_2 = player2
+    @board = board
+  end
+
+  WIN_COMBINATIONS = [
+    [0,1,2], #top row
+    [3,4,5], #middle row
+    [6,7,8], #bottom row
+    [0,3,6], #left column
+    [1,4,7], #middle column
+    [2,5,8], #right column
+    [0,4,8], #top left to bottom right
+    [6,4,2]  #bottom left to top right
+  ]
+
+  def current_player
+    (self.board.turn_count.even? ? self.player_1 : self.player_2)
+  end
+
+  def over?
+    (self.draw? || self.won? ? true : false)
+  end
+
+  def full?
+    @board.cells.each do |token|
+      if token == " "
+        return false
+      end
+    end
+  end
+
+  def draw?
+    if full? && !won?
+      return true
+    elsif !won? && !full?
+      return false
+    else
+      return false
+    end
+  end
+
+  def won?
+    winner = false
+
+    WIN_COMBINATIONS.each do |combo|
+      if @board.cells[combo[0]] == "X" && 
+         @board.cells[combo[1]] == "X" && 
+         @board.cells[combo[2]] == "X" || 
+         @board.cells[combo[0]] == "O" && 
+         @board.cells[combo[1]] == "O" && 
+         @board.cells[combo[2]] == "O"
+          winner = [combo[0],combo[1],combo[2]]
+      end
+    end
+
+    return winner
+  end
+
+  def winner
+    if won? == false
+      return nil
+    else 
+      @board.cells[won?[0]]
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.